src/runtime/nitro/plugins/40-cspSsrNonce.ts

```diff
@@ -17,18 +18,32 @@ export default defineNitroPlugin((nitroApp) => {
     return
   }
 
+  // Genearate a 16-byte random nonce for each request.
   nitroApp.hooks.hook('request', (event) => {
+    if (isIslandRequest(event)) {
```

Comment:
Is it more simple to use routeRules instead of using `.startsWith` ?

Reply:
Could you give an example? Comparing the request path is the only way I know to check if a request is for rendering an island component.

Reply:
We should be able know to augment routeRules

```ts
defineNuxtConfig({
  routeRules: { '/__nuxt_island/**' : {isIsland: true }}
})
```

or we could also mark the event being an island request with an island only plugin (using the env property with the object plugin syntax)

The issue is that this hook gets called at each request so we should make it as performant as possible. Boolean check is always faster than `startsWith()`
![image](https://github.com/user-attachments/assets/51fadde2-656a-4ea3-91af-1a327163ea94)

Reply:
I don't think that a single `startsWith()` call in the Nuxt request pipeline will have a relevant performance impact.
Using route rules in this case makes the code more complicated in my opinion and requires to augment the TypeScript types correctly.
Route rules need to be matched by `event.path` too, which is likely using a regular expression under the hood, which is slower than a call to `string.startsWith()`.

Reply:
We could instead simply check if the nonce already exist, to not generate it twice. Generating it twice could lead to issues, where the nonce differs at different times of the request pipeline.

Reply:
I removed the `isIslandRequest` util and replaced it with a simple check if the nonce already exists.